packages/core/src/pattern.mjs

```diff
@@ -1132,6 +1143,7 @@ Pattern.prototype.__optionSets = function (optionName) {
  * @return {Pattern} this - The Pattern instance
  */
 Pattern.prototype.__pack = function () {
+  this.__runHooks('preLayout')
```

Comment:
I wanted hooks that got called internally by `__pack` because as it was `preRender` was being called *before* `__pack` in `getRenderProps` and *after* `__pack` in `render`, and I needed a consistent order of operations